src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/get_interior_vehicle_data_request_test.cc

```diff
@@ -130,113 +145,306 @@ class GetInteriorVehicleDataRequestTest
           mobile_message);
   command->Run();
+  application_manager::event_engine::Event event(
+      hmi_apis::FunctionID::RC_GetInteriorVehicleData);
+  event.set_smart_object(*hmi_response);
+  command->on_event(event);
 }
 
 TEST_F(GetInteriorVehicleDataRequestTest,
-       OnEvent_ValidHmiResponse_ExpectSuccessfullResponseSentToMobile) {
+       Execute_ExpectMessageNotSentToHMI_TwoApps_OneUnsubscribed) {
+  // Arrange
+  auto module_type = mobile_apis::ModuleType::RADIO;
+
+  MessageSharedPtr mobile_message = CreateBasicMessage();
+  (*mobile_message)[application_manager::strings::msg_params]
+                   [message_params::kModuleType] = module_type;
+  (*mobile_message)[application_manager::strings::msg_params]
+                   [message_params::kSubscribe] = false;
+
+  apps_.insert(mock_app_);
+  apps_.insert(mock_app2_);
+  rc_app_extention_->SubscribeToInteriorVehicleData(enums_value::kRadio);
+  rc_app_extention2_->SubscribeToInteriorVehicleData(enums_value::kRadio);
+
+  smart_objects::SmartObject radio_data;
+  radio_data[message_params::kBand] = enums_value::kAM;
+
+  ON_CALL(mock_policy_handler_, CheckModule(_, enums_value::kRadio))
+      .WillByDefault(Return(true));
+  ON_CALL(mock_hmi_capabilities_, rc_capability())
+      .WillByDefault(Return(nullptr));
+  ON_CALL(app_mngr_, applications()).WillByDefault(Return(apps_da_));
+
+  application_manager::SharedPtr<
+      rc_rpc_plugin::commands::GetInteriorVehicleDataRequest> command =
+      CreateRCCommand<rc_rpc_plugin::commands::GetInteriorVehicleDataRequest>(
+          mobile_message);
+
+  // Expectations
+  EXPECT_CALL(mock_interior_data_cache_, Contains(enums_value::kRadio))
+      .WillOnce(Return(true));
+  EXPECT_CALL(mock_interior_data_cache_, Retrieve(enums_value::kRadio))
+      .WillOnce(Return(radio_data));
+
+  EXPECT_CALL(mock_rpc_service_, ManageHMICommand(_)).Times(0);
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_apis::Result::SUCCESS), _))
+      .WillOnce(Return(true));
+
+  // Act
+  command->Run();
+
+  // Assert
+  EXPECT_FALSE(
+      rc_app_extention_->IsSubscibedToInteriorVehicleData(enums_value::kRadio));
+}
+
+TEST_F(
+    GetInteriorVehicleDataRequestTest,
+    Execute_CapabilitiesValidationFailed_ExpectMessageNotSentToHMI_AndFalseSentToMobile) {
   // Arrange
   MessageSharedPtr mobile_message = CreateBasicMessage();
   NsSmartDeviceLink::NsSmartObjects::SmartObject& msg_params =
       (*mobile_message)[application_manager::strings::msg_params];
+  msg_params[message_params::kModuleType] = mobile_apis::ModuleType::RADIO;
+  application_manager::SharedPtr<
+      rc_rpc_plugin::commands::GetInteriorVehicleDataRequest> command =
+      CreateRCCommand<rc_rpc_plugin::commands::GetInteriorVehicleDataRequest>(
+          mobile_message);
+
+  ON_CALL(mock_policy_handler_, CheckModule(_, _)).WillByDefault(Return(true));
+  // Used empty rc capabilities
```

Comment:
@LuxoftAKutsan Did not get the idea of this comment... It is related to..???

Reply:
@AByzhynar it is not clear that rc_capabilities will be used from ON_CALLS frin constructir and it is empty by default. SO I added it as a comment

Reply:
@LuxoftAKutsan Confusing comment or place for it... May be add it as condition to test case name?